File: FoodieApp(FE)/src/app/models/restaurant.ts
export interface Restaurant {
    _id: string;
    name: string;
    address: string;
    city: string;
    state: string;
    rating: number;
    contact: Contact;
    cuisine: string[];
    menu: Menu[];
    reviews: Review[];
    features: Features;
    images: string;
    social_media: SocialMedia;
    payment_methods: string[];
    specials: Specials;
    health_safety: HealthSafety;
    favourite: boolean;
  }
  
  export interface Contact {
    
  }
  
  export interface Menu {
    name: string;
    price: number;
    rating: number;
    image: string;
  }
  export interface Review {
   
  }
  
  export interface Features {
   
  }
  
  export interface SocialMedia {
  
  }
  
  export interface Specials {
 
  }
  
  export interface HealthSafety {
   
  }

File: FoodiesHub(BE)/restaurants_api/src/main/java/com/major/restaurants_api/model/Restaurant.java
package com.major.restaurants_api.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "restaurant")
public class Restaurant {
    @Id
    private String _id;
    private String name;
    private String address;
    private String city;
    private String state;
    private double rating;
//    private Coordinates coordinates;
    private Contact contact;
    private List<String> cuisine;
    private List<Menu> menu;
    private List<Review> reviews;
    private Features features;
    private String images;
    private SocialMedia social_media;
    private List<String> payment_methods;
    private Specials specials;
    private HealthSafety health_safety;
}
